Dockerfile

```diff
@@ -8,12 +8,14 @@ ENV INSTALLED_SEMGREP_VERSION=0.39.1
 
 RUN apk add --no-cache --virtual=.build-deps build-base libffi-dev openssl-dev &&\
     apk add --no-cache --virtual=.run-deps bash git less libffi openssl &&\
-    pip install --no-cache-dir poetry==1.0.10 &&\
+    # Need to pin cryptography version to avoid Rust compiler dependency
+    pip install --no-cache-dir cryptography==3.3.2 poetry==1.0.10 &&\
     pip install --no-cache-dir pipx &&\
     pipx install semgrep==${INSTALLED_SEMGREP_VERSION} &&\
-    poetry config virtualenvs.create false &&\
-    # Don't install dev dependencies or semgrep-agent
-    poetry install --no-dev --no-root &&\
+    poetry config virtualenvs.create false
+
+# Don't install dev dependencies or semgrep-agent
+RUN poetry install --no-dev --no-root &&\
```

Comment:
We shouldn't split into two `RUN` instructions; that negates all our disk space saving measures right below this line. If the above `RUN` finishes before uninstalling stuff, then `poetry`, `libffi-dev`, etc. will all be committed to the final image. Compressed image size went from [86 MB](https://hub.docker.com/layers/returntocorp/semgrep-agent/v1/images/sha256-fd030566af43ead16e15c81740b38da076f0563ec8ae69ee70ed1fc8519b0e87?context=explore) to [206 MB](https://hub.docker.com/layers/returntocorp/semgrep-agent/pr-197/images/sha256-f2f02b2b14adb9440154e8b1cf59b2287e79d443598998f9f3e2b7190a84797c?context=explore).